### ISSUE TITLE
Move handling of translations (itext) to the client?

### ISSUE THREAD

**Issue by MartijnR** [enketo/enketo#727]
Only after client XPath performance has been improved dramatically.

This makes sense for the following reasons:
- the current mechanism has bugs because extracting the itext id from a string is not done well (since don't have a regexp functions at our disposal in enketo-transformer)
- often option lists are used multiple times and currently copied multiple times which can cause the # of DOM elements to explode in some forms.
- the current mechanism is not XForms-spec-compliant, because you cannot use itext in a calculation for example (it's not implemented as an XPath function)
- XSLT transformation could be up to 85% quicker, which would have a noticeable impact on loading long forms for the first time and for loading previews (although the latter could be cached too, if it is likely previews are loaded multiple times without form changes in between). However, this impact would depend on whether the client can handle itext faster (I think it can, especially if the native XPath processor can be used).
- ~~because of the first point it would become easier and more elegant to add support for external instances (itemsets).~~

Not a priority.

Client initial loading performance may be an issue, though reducing the # of DOM elements would have a somewhat positive impact.

Language switching will become slower as well, but that's no big deal at all.
